src/block/block_tiered.c

```diff
@@ -74,8 +74,10 @@ __wt_block_tiered_newfile(WT_SESSION_IMPL *session, WT_BLOCK *block)
     if (session->bucket_storage != NULL && block->logid != 0) {
         storage_source = session->bucket_storage->storage_source;
         WT_ASSERT(session, storage_source != NULL);
-        WT_ERR(storage_source->ss_flush(
-          storage_source, &session->iface, session->bucket_storage->file_system, filename, NULL));
+        WT_ERR(storage_source->ss_flush(storage_source, &session->iface,
+          session->bucket_storage->file_system, filename, filename, NULL));
+        WT_ERR(storage_source->ss_flush_finish(storage_source, &session->iface,
+          session->bucket_storage->file_system, filename, filename, NULL));
```

Comment:
For both of these it is strange to see sending in `filename` twice. The args are `source` and `object`. I think if we really need the same thing twice it needs a comment about why? Or if we really don't need it twice, we can just use the one name for multiple purposes in those functions and eliminate one. Are they ever different? Wouldn't it be the same name just moved/flushed/copied from one location to the other?

Reply:
Yes, I have the ticket for the internal thread and spinning off stuff on my plate. Thanks for clarifying.